napari/_qt/dialogs/qt_notification.py

```diff
@@ -301,17 +295,46 @@ def sizeHint(self):
         )
 
     @classmethod
-    def from_exception(cls, exception: BaseException) -> 'NapariNotification':
-        """Create a NapariNotifcation dialog from an exception object."""
-        # TODO: this method could be used to recognize various exception
-        # subclasses and populate the dialog accordingly.
-        extra_msg = (
-            "\nYou can start napari with NAPARI_CATCH_ERRORS=0 or "
-            "NAPARI_EXIT_ON_ERROR=1 to find more about this error"
+    def from_notification(cls, notification):
+        actions = notification.actions
+
+        if not actions and isinstance(notification, ErrorNotification):
+            # TODO: this should be moved
+
+            exc = notification.exception
+
+            def show_tb(parent):
+                tbdialog = QDialog(parent=parent.parent())
+                tbdialog.setModal(True)
+                tbdialog.setLayout(QVBoxLayout())
+                try:
+                    format_exc_info = get_tb_formatter()
+                except Exception:
+                    import traceback
+                    traceback.print_tb()
+                    raise
+                info = (exc.__class__, exc, exc.__traceback__)
+                text = QTextEdit()
+                text.setHtml(format_exc_info(info, as_html=True))
+                text.setReadOnly(True)
+                print()
+                layout = tbdialog.layout()
```

Comment:
This line triggers a:
```
wrapped C/C++ object of type QDialog has been deleted
Traceback (most recent call last):
  File "/Users/mbussonnier/dev/napari/napari/_qt/dialogs/qt_notification.py", line 281, in <lambda>
    btn.clicked.connect(lambda: callback(self))
  File "/Users/mbussonnier/dev/napari/napari/_qt/dialogs/qt_notification.py", line 321, in show_tb
    layout = tbdialog.layout()
RuntimeError: wrapped C/C++ object of type QDialog has been deleted
```

It's crazy as it does work if i uncomment on the  `format_exc_info = get_tb_formatter()` on line 29.

What I _believe_ is happening is the dialog is parented to the notification toast, that get dismissed when clicking on "traceback" and crash napari; but I'm not sure and can't figure out how to fix it. 

@goanpeca maybe if you have 30 minutes, you might help me to debug ?

Reply:
Just tried and couldn't repeat it... can you tell me how you're triggering the exception and how you're running napari?

Reply:
As discussed on the call with @Carreau these errors usually refer to C/C++ objects being destroyed when there is no (permanent) reference to them kept in python. Also moving the nested func to a method might make it easier to debug and cleaner to keep a ref to either the cls or a self (if it is instantiated)

Reply:
Yep, that's fine... still would like to be able to repeat though

Reply:
I've tried to disable gc,to no avail; so I'm unsure. 
with current branch if you `export DEBUG_NAPARI_NOTIFICATION=1` then two extra buttons in the interface can trigger an exception and a warning. Trying to click "traceback" crash napari, here is a video:
![crsh](https://user-images.githubusercontent.com/335567/107416854-cf128400-6ac9-11eb-81b9-42cb476d35cb.gif)

Reply:
it doesn't really change the necessary action (which we all agree is to just maintain a reference)... but fwiw, I still don't get that error! 🤷‍♂️

![Untitled](https://user-images.githubusercontent.com/1609449/107675316-f7cf8080-6c65-11eb-9389-d6b8ba6f68fa.gif)

still though, I feel like the point of this PR shouldn't be to focus on the qt_notifications module, but rather to setup an API around the notification manager and the events _it_ emits... leaving the visual display in various notification front ends for another PR

Reply:
Sure, but it's hard to test if it crashes, and make sure it does the right things. I'll strip down the things that do pop up the dialog then.

Reply:
> but it's hard to test if it crashes, and make sure it does the right things. 

maybe create a simple terminal notifier?